### DISCUSSION TITLE
Where is string_pool used?

### DISCUSSION THREAD

**Discussion by qua3k** [rui314/mold#1045] [Q&A] [answered]
Hi, I’m looking at implementing a linker of my own and I’ve gotten to the save_string function and see that it adds the string to the string_pool, but doing [a code search over string_pool](https://github.com/search?q=repo%3Arui314%2Fmold%20string_pool&type=code) doesn’t reveal it being used anywhere else.

**Comment by rui314** [comment, marked as answer]
We use the vector to keep strings alive for the entire execution of the linker.

**Reply by qua3k** [reply]
Oh I understand, this is part of the not freeing unnecessarily. Thanks!
